### ISSUE TITLE
False positive: `too-many-locals` (R0914) counts PEP 695 type parameters as local variables

### ISSUE THREAD

**Issue by Synvoya** [PyCQA/pylint#11136]
### Bug description

`too-many-locals` (R0914) counts PEP 695 type parameters (`def f[T1, T2, ...]`) as local variables, so a generic function with many type parameters falsely trips the check even when it has almost no real locals.

```python
def build[T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16](x: T1) -> T1:
    return x
```

This function has exactly **one** real local (`x`), but pylint counts the 16 type parameters too.

### Command used

```shell
pylint --disable=all --enable=too-many-locals f.py
```

### Pylint output

```
f.py:1:0: R0914: Too many local variables (17/15) (too-many-locals)
```

### Expected behavior

No message. PEP 695 type parameters are type-system constructs, not runtime local variables. pylint already excludes them from other checkers (e.g. `undefined-variable`, `used-before-assignment`); `too-many-locals` should do the same.

### Root cause

`pylint/checkers/design_analysis.py` computes `locnum = len(node.locals) - ignored_args_num`, but astroid places PEP 695 type-param names into `node.locals`, over-counting by the number of type parameters.

### Pylint version

```
pylint 4.x (main branch), astroid 4.x, Python 3.12
```
